src/Servers/Kestrel/Transport.MsQuic/src/Internal/MsQuicStream.cs

```diff
@@ -390,20 +401,45 @@ public unsafe void EnableReceive()
             SetParam(QUIC_PARAM_STREAM.RECEIVE_ENABLED, buffer);
         }
 
+        private unsafe long GetStreamId()
+        {
+            Span<byte> localByteArray = stackalloc byte[8];
```

Comment:
You can avoid the `Span<byte>` and `fixed` here.
```c#
private unsafe long GetStreamId()
{
    byte* ptr = stackalloc byte[sizeof(long)];
    var buffer = new QuicBuffer
    {
        Length = sizeof(long),
        Buffer = ptr
    };
    GetParam(QUIC_PARAM_STREAM.ID, buffer);
    return BitConverter.ToInt64(localByteArray);
}
```

If it would be on a really hot path, I would write this as
```c#
private unsafe long GetStreamId()
{
    long tmp;
    var buffer = new QuicBuffer
    {
        Length = sizeof(long),
        Buffer = &tmp
    };
    GetParam(QUIC_PARAM_STREAM.ID, buffer);
    return BitConverter.ToInt64(localByteArray);
}
```
(Ideally the JIT would generate the same code for both. JIT treats small constants (< 32 byte) as normal locals, but introduces a safety to check stack-corruption)

Reply:
How do you recommend going from byte* to Span<byte>? BitConverter.ToInt64 requires a Span<byte>.

Reply:
Hah leave it then

Reply:
`BitConverter.ToInt64` is actually `*(long*)ptr`. As it's unsafe anyway you could write
```c#
return *(long*)ptr;
```

(Ah, I see my copy & paste error in the snippet above).